src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/events/InboundEventsServiceTest.kt

```diff
@@ -0,0 +1,89 @@
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.TestLogAppender
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.aValidPrisonNumber
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.TimelineEvent
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.TimelineEventType
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.service.TimelineService
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.service.events.aValidCiagInductionCreatedEvent
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.service.events.aValidCiagInductionUpdatedEvent
+import java.time.Instant
+import java.util.UUID
+
+@ExtendWith(MockitoExtension::class)
+internal class InboundEventsServiceTest {
+
+  @InjectMocks
+  private lateinit var inboundEventsService: InboundEventsService
+
+  @Mock
+  private lateinit var timelineService: TimelineService
+
+  @Captor
+  private lateinit var timelineEventCaptor: ArgumentCaptor<TimelineEvent>
+
+  @Test
+  fun `should process CIAG induction created event`() {
+    // Given
+    val prisonNumber = aValidPrisonNumber()
+    val inductionCreatedEvent = aValidCiagInductionCreatedEvent()
+
+    // When
+    inboundEventsService.process(inductionCreatedEvent)
+
+    // Then
+    verify(timelineService).recordTimelineEvent(eq(prisonNumber), capture(timelineEventCaptor))
+    assertThat(timelineEventCaptor.value.sourceReference).isEqualTo(inductionCreatedEvent.reference())
+    assertThat(timelineEventCaptor.value.eventType).isEqualTo(TimelineEventType.INDUCTION_CREATED)
+    assertThat(timelineEventCaptor.value.prisonId).isEqualTo(inductionCreatedEvent.prisonId())
+    assertThat(timelineEventCaptor.value.actionedBy).isEqualTo(inductionCreatedEvent.userId())
+  }
+
+  @Test
+  fun `should process CIAG induction updated event`() {
+    // Given
+    val prisonNumber = aValidPrisonNumber()
+    val inductionUpdatedEvent = aValidCiagInductionUpdatedEvent()
+
+    // When
+    inboundEventsService.process(inductionUpdatedEvent)
+
+    // Then
+    verify(timelineService).recordTimelineEvent(eq(prisonNumber), capture(timelineEventCaptor))
+    assertThat(timelineEventCaptor.value.sourceReference).isEqualTo(inductionUpdatedEvent.reference())
+    assertThat(timelineEventCaptor.value.eventType).isEqualTo(TimelineEventType.INDUCTION_UPDATED)
+    assertThat(timelineEventCaptor.value.prisonId).isEqualTo(inductionUpdatedEvent.prisonId())
+    assertThat(timelineEventCaptor.value.actionedBy).isEqualTo(inductionUpdatedEvent.userId())
+  }
+
+  @Test
+  fun `should fail to process unrecognised event`() {
+    // Given
+    val unrecognisedEvent = UnrecognisedInboundEvent()
+
+    // When
+    inboundEventsService.process(unrecognisedEvent)
+
+    // Then
+    assertThat(TestLogAppender.hasLog("Unsupported event uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service.events.UnrecognisedInboundEvent", Level.WARN)).isTrue
```

Comment:
This testlogappender stuff reminds me of VCA and AS's code (and I never liked in there! 🤣 )
I think the reason I dont like making business process assertions based on log messages is that logging can change - perhaps we remove the logging altogether from the service class, in which case this test breaks

In terms of the business process I guess what we are saying here is that if an unsupported event type is sent then no calls to the timeline service are made?
Would it be simpler and more readable to do a verify no calls to timeline service mock instead?

Reply:
> I think the reason I dont like making business process assertions based on log messages is that logging can change

But isn't that partly the point of any test - to make sure someone doesn't change it unwittingly?

Happy to verify no calls to the service mock instead, though I don't think it's as thorough.

Reply:
Removed :)